### DISCUSSION TITLE
Is there a better approach to implementing this

### DISCUSSION THREAD

**Discussion by haztakki** [filamentphp/filament#3507] [Help] [answered]
Hello!
Excuse the poor topic title, but I was wondering if there was a better way to approach this. Basically, I have a validation group, each group has a validation type and if necessary, a validation value. For example, the "min" validation type requires a value (e.g. 10), the "required" validation type doesn't. Here is a screenshot to give you an idea of how it works:

<img width="540" alt="Screenshot 2022-08-15 at 15 06 40" src="https://user-images.githubusercontent.com/20573863/184635584-1cbfa957-97fe-41aa-b732-0285c458cd1a.png">

So far (it is still very WIP), everything is setup and working perfectly, at least on the UI side of things. _I will_ manipulate the data before saving to the database, probably using **mutateFormDataBeforeCreate**. I just haven't gotten to that yet.

I know it is kind of a broad question, but is there a better way of achieving this behaviour?

```php
->schema([
                // ...
                Repeater::make(name: 'validation')
                    ->schema([
                        Select::make(name: 'rule')
                            ->name(name: 'rule')
                            ->options(options: []) // Validation rules
                            ->required()
                            ->reactive(),

                        TextInput::make(name: 'value')
                            ->reactive()
                            ->disabled(fn(Closure $get) => in_array(needle: $get('rule'), haystack: static::$valueRequiredValidationTypes, strict: true)),
                    ])
                    ->columns()
            ]);
```

One thing that comes to mind, are the various edge cases like, making sure that each validation rule value is the correct type (string, numeric, etc).

**Comment by ryangjchandler** [comment, marked as answer]
You can conditionally add validation rules, so could change how the value field gets validated based on the rule being used.

**Reply by haztakki** [reply]
> You can conditionally add validation rules, so could change how the value field gets validated based on the rule being used. 

Yeah, that's what I was thinking. Other than that, do you think the approach is okay? I'm new to Filament so just seeing if there is a better alternative. Thanks for your reply.

**Reply by danharrin** [reply]
The approach looks ok, as Ryan said you can use conditional validation rules with closures to achieve the validation based on the datatype. Check the Advanced section of the forms docs to read all possibilities.

**Reply by haztakki** [reply]
Thanks both of you, appreciate the feedback.
